src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -563,29 +550,28 @@ def get_subscription_id(self, subscription=None):  # take id or name
         return self.get_subscription(subscription)[_SUBSCRIPTION_ID]
 
     @staticmethod
-    def _try_parse_msi_account_name(account):
-        msi_info, user = account[_USER_ENTITY].get(_ASSIGNED_IDENTITY_INFO), account[_USER_ENTITY].get(_USER_NAME)
-
-        if user in [_SYSTEM_ASSIGNED_IDENTITY, _USER_ASSIGNED_IDENTITY]:
-            if not msi_info:
-                msi_info = account[_SUBSCRIPTION_NAME]  # fall back to old persisting way
```

Comment:
In versions <= 2.0.50 (November 6, 2018), the subscription name is used to denote the managed identity ID type. This is super confusing, so `_ASSIGNED_IDENTITY_INFO` was added in https://github.com/Azure/azure-cli/pull/7744 and these lines are added as an adaptor. However, such logic is still difficult to maintain and can easily leads to unwanted code path.

Reply:
If we remove this adaptor, we can add such history notes:

[BREAKING CHANGE] No long be compatible with login profile created by `az login --identity` of Azure CLI versions <= 2.0.50. If you are updating from those versions, please run `az login --identity` again to refresh the login profile.